libraries/aws/provisioning/test/unit/aws_iot_tests_provisioning_serializer.c

```diff
@@ -141,6 +228,71 @@ TEST( Provisioning_Unit_Serializer, TestSerializeCreateKeysAndCertificatePayload
     AwsIotProvisioning_FreePayload( pSerializationBuffer );
 }
 
+/**
+ * @brief Tests that the CSR payload serializer calculates the serialization size, and populates the passed output
+ * parameter when a serialization buffer is not passed to it.
+ */
+TEST( Provisioning_Unit_Serializer, TestCalculateCertFromCsrPayloadSize )
+{
+    size_t bufferSizeNeeded = 0;
+
+    /* Test the serializer function. */
+    TEST_ASSERT_EQUAL( true,
+                       _AwsIotProvisioning_CalculateCertFromCsrPayloadSize( _testCsrString,
+                                                                            strlen( _testCsrString ),
+                                                                            &bufferSizeNeeded ) );
+    /* Make sure that the output parameter has been populated with the serialization size. */
+    TEST_ASSERT_EQUAL( sizeof( _expectedSerialization ), bufferSizeNeeded );
+}
+
+/**
+ * @brief Tests that the CSR payload serializer populates the buffer with the serialized payload, when
+ * the passed buffer is valid.
+ */
+TEST( Provisioning_Unit_Serializer, TestSerializeCreateCertFromCsrPayloadWithBuffer )
+{
+    size_t reserveOffset;
+
+    /* Allocate buffer that will be used for serialization.
+     * Note: We will allocate more space than the expected serialization size to perform
+     * buffer overrun checks after the serialization operation.  */
+    size_t serializationSize = sizeof( _expectedSerialization );
+    uint8_t * testBuffer = _allocateBufferMemoryWithBufferOverrunReserve(
+        serializationSize, &reserveOffset );
```

Comment:
Is this spacing a result of uncrustify? I thought when we split functions we only have one parameter per line

Reply:
Nope, it was manual. I will keep update to parameters on different lines.

Reply:
Ideally, uncrustify should have automatically separated the parameters on new lines when formatting this code text.